### PR TITLE
Working version to fixed mode-line toggle issue.

### LINKED ISSUE #40: Bug toggling `feebleline-mode`.
I don't know what is happening but after I toggled certain times of `feebleline-mode` then this appeared.

Image 1

### PR DESCRIPTION
**Pull request by jcs090218**
As title, this is just a working version for now. If anyone have knowledge about walk through windows, please improve it!

Fixed #40!

### CODE REVIEW

**Comment by tautologyclub** [comment]
Don't think that's the cleanest way to do it -- I promise to take a look as soon as I have time.

**Comment by jcs090218** [comment]
@tautologyclub No worries, I just have the PR and probably will improve it later on! And most importantly this give others the solution!

File: feebleline.el [hunk 1]
```diff
@@ -214,6 +216,25 @@ Returns a pair with desired column and string."
   (with-current-buffer feebleline--minibuf
     (erase-buffer)))
 
+
+(defun feebleline--count-windows ()
+  "Total window count."
+  (let ((count 0))
+    (dolist (fn (frame-list))
+      (setq count (+ (length (window-list fn)) count)))
+    count))
```
Comment (jcs090218) [inline feebleline.el]: Just to make sure, this function count all windows across all frames. :D That's why I don't use normal count windows, and I don't know if there is a default way of doing this... :(

File: feebleline.el [hunk 2]
```diff
@@ -214,6 +216,25 @@ Returns a pair with desired column and string."
...
+
+(defun feebleline--walk-through-windows (fnc)
+  "Walk through all the windows once and execute callback FNC."
+  (save-selected-window
+    (let ((index 0))
+      (while (< index (feebleline--count-windows))
+        (when fnc
+          (funcall fnc))
+        (other-window 1 t)
+        (setq index (+ index 1))))))
```
Comment (jcs090218) [inline feebleline.el]: This function just need a function pointer `fnc`, and call it every time when the window changed.

**Review by jcs090218 — COMMENTED** [review]
Some implementation explanation.

**Comment by kevinboulain** [comment]
I'm not sure about the fix itself, but you can replace the two functions you introduced with: `(mapc (lambda (frame) (mapc #'print (window-list frame))) (frame-list))` (replace `print`).

**Comment by jcs090218** [comment]
@ether42 Thanks for the method! Just updated! I don't know if this is a proper fix but this fixed the bug for me! :( I used to work with two windows; if this doesn't get fixed then this is a bit awkward when using this plugin.

**Edit:** And of course, if anyone has another solution and doesn't do the ugly walk through all windows work. This will be awesome! :D

**Comment by kevinboulain** [comment]
I'm sorry, I should have been more precise. Since you don't use `window` you are doing all of the `setq` in the current one.

By using `with-selected-window` you can switch to the window before performing an action: 
```elisp
(mapc (lambda (frame)
        (mapc (lambda (window)
                (with-selected-window window
                  (insert "hello world")))
              (window-list frame)))
      (frame-list))
```

**Comment by jcs090218** [comment]
No problem! Was also my fault that I should had checked the method carefully. :D

**Edit:** Fixed!

**Comment by kevinboulain** [comment]
I just saw that I use `(walk-windows FUN &optional MINIBUF ALL-FRAMES)` in my configuration, I'm terribly sorry for the back and forth but this one should be much better.

**Comment by jcs090218** [comment]
Nice! This is much cleaner right now! :D

**Comment by jcs090218** [comment]
@tautologyclub @endian-benjamin Do you think this patch can be merged?

**Comment by jcs090218** [comment]
@tautologyclub @endian-benjamin Ping.

**Comment by endian-benjamin** [comment]
oh wow, apologies - completely forgot about this due to vacations and all. Will get to it very soon, I promise

**Comment by tautologyclub** [comment]
Nice work

**Pull request merged**